code/create_survey_geojson.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import json
+import re
+from pathlib import Path
+from typing import Iterable, Optional
+
+import pandas as pd
+
+# path to data directory
+DPATH_DATA = Path(__file__).parent.parent / "data"
+
+# path to assets directory (for GeoJSON data input/output)
+DPATH_ASSETS = Path(__file__).parent / "assets"
+
+# path to generic US states GeoJSON file
+FPATH_STATES_JSON_DEFAULT = DPATH_ASSETS / "us_states.json"
+
+# path to output GeoJSON file
+FPATH_OUT_DEFAULT = DPATH_ASSETS / "survey_states.json"
+
+
+def create_survey_geojson(
+    survey_states_and_clusters: Iterable[str],
+    fpath_states_json: Path | str = FPATH_STATES_JSON_DEFAULT,
+    fpath_out: Optional[Path | str] = FPATH_OUT_DEFAULT,
+) -> dict:
+    """Create a GeoJSON file for the states and state clusters used in the survey.
+
+    Parameters
+    ----------
+    survey_states_and_clusters : Iterable[str]
+        State and state cluster names to include in the GeoJSON file. State
+        cluster names should be in the format "State1, State2, ... (Cluster X)".
+    fpath_states_json : Path | str, optional
+        Path to GeoJSON file with standard US states.
+        By default, will use the one that was
+        downloaded from https://github.com/PublicaMundi/MappingAPI/blob/master/data/geojson/us-states.json.
+    fpath_out : Path | str, optional
+        Path to output GeoJSON file.
+
+    Returns
+    -------
+    dict
+        GeoJSON of the survey states and state clusters.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if state clusters cannot be parsed.
+    """
+    # use default paths if none provided
+    if fpath_states_json is None:
+        fpath_states_json = FPATH_STATES_JSON_DEFAULT
+    if fpath_out is None:
+        fpath_out = FPATH_OUT_DEFAULT
+
+    fpath_states_json = Path(fpath_states_json)
+    fpath_out = Path(fpath_out)
+
+    # load GeoJSON file
+    states_json = json.loads(fpath_states_json.read_text())
+
+    # create mapping for easy access to reference state information
+    state_info_map = {
+        info["properties"]["name"]: info for info in states_json["features"]
+    }
+
+    # split into states and clusters
+    survey_states = [
+        state for state in survey_states_and_clusters if "Cluster" in state
+    ]
+    survey_clusters = [
+        state
+        for state in survey_states_and_clusters
+        if state not in survey_states
+    ]
+
+    survey_state_info = []
```

Comment:
I forgot/got a little confused by what this variable was by the time I reached the bottom of the following for loop. How about making the name even more explicit, something like

```suggestion
    # Array of feature dictionaries representing geometries of states or custom state clusters used in the survey
    survey_regions_geo_features = []
```

or similar?